The operator-sdk can't be used with vendored kubevirt

**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
/kind enhancement


**What happened**:

We have 
```
func init() {
	flag.StringVar(&kubeconfig, "kubeconfig", "", "absolute path to the kubeconfig file")
	flag.StringVar(&master, "master", "", "master url")
}
```

in

https://github.com/kubevirt/kubevirt/blob/master/pkg/kubecli/kubecli.go#L55

The flags here are also defined in the operator-sdk and they clash:

```
log is DEPRECATED and will be removed in a future version. Use logs instead.
kubevirt-vm-operator flag redefined: kubeconfig
panic: kubevirt-vm-operator flag redefined: kubeconfig

goroutine 1 [running]:
flag.(*FlagSet).Var(0xc00007a180, 0x18cf060, 0x27590a0, 0x16aa419, 0xa, 0x16ed85b, 0x37)
        /usr/lib/golang/src/flag/flag.go:850 +0x4af
flag.StringVar(...)
        /usr/lib/golang/src/flag/flag.go:759
github.com/masayag/kubevirt-vm-operator/vendor/sigs.k8s.io/controller-runtime/pkg/client/config.init.0()
        /home/masayag/work/go/src/github.com/masayag/kubevirt-vm-operator/vendor/sigs.k8s.io/controller-runtime/pkg/client/config/config.go:38 +0x98
```

**What you expected to happen**:

Setting these flags should be optional. Let's keep this in mind for #2297.

**How to reproduce it (as minimally and precisely as possible)**:


**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`):
- Kubernetes version (use `kubectl version`):
- VM or VMI specifications:
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

--- a/pkg/service/BUILD.bazel
+++ b/pkg/service/BUILD.bazel
@@ -5,5 +5,8 @@ go_library(
     srcs = ["service.go"],
     importpath = "kubevirt.io/kubevirt/pkg/service",
     visibility = ["//visibility:public"],
-    deps = ["//vendor/github.com/spf13/pflag:go_default_library"],
+    deps = [
+        "//staging/src/kubevirt.io/client-go/kubecli:go_default_library",
+        "//vendor/github.com/spf13/pflag:go_default_library",
+    ],
 )

--- a/pkg/service/service.go
+++ b/pkg/service/service.go
@@ -25,7 +25,13 @@ import (
 	"strconv"
 
 	flag "github.com/spf13/pflag"
+
+	"kubevirt.io/client-go/kubecli"
 )
+
+func init() {
+	kubecli.Init()
+}
 
 type Service interface {
 	Run()

--- a/pkg/virt-api/rest/subresource_test.go
+++ b/pkg/virt-api/rest/subresource_test.go
@@ -49,6 +49,8 @@ import (
 )
 
 var _ = Describe("VirtualMachineInstance Subresources", func() {
+	kubecli.Init()
+
 	var server *ghttp.Server
 	var backend *ghttp.Server
 	var request *restful.Request

--- a/staging/src/kubevirt.io/client-go/kubecli/kubecli.go
+++ b/staging/src/kubevirt.io/client-go/kubecli/kubecli.go
@@ -54,8 +54,12 @@ var once sync.Once
 // Init adds the default `kubeconfig` and `master` flags. It is not added by default to allow integration into
 // the different controller generators which normally add these flags too.
 func Init() {
-	flag.StringVar(&kubeconfig, "kubeconfig", "", "absolute path to the kubeconfig file")
-	flag.StringVar(&master, "master", "", "master url")
+	if flag.CommandLine.Lookup("kubeconfig") == nil {
+		flag.StringVar(&kubeconfig, "kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	if flag.CommandLine.Lookup("master") == nil {
+		flag.StringVar(&master, "master", "", "master url")
+	}
 }
 
 func GetKubevirtSubresourceClientFromFlags(master string, kubeconfig string) (KubevirtClient, error) {

--- a/staging/src/kubevirt.io/client-go/kubecli/kubecli.go
+++ b/staging/src/kubevirt.io/client-go/kubecli/kubecli.go
@@ -51,7 +51,9 @@ var (
 var virtclient KubevirtClient
 var once sync.Once
 
-func init() {
+// Init adds the default `kubeconfig` and `master` flags. It is not added by default to allow integration into
+// the different controller generators which normally add these flags too.
+func Init() {
 	flag.StringVar(&kubeconfig, "kubeconfig", "", "absolute path to the kubeconfig file")
 	flag.StringVar(&master, "master", "", "master url")
 }

--- a/tests/utils.go
+++ b/tests/utils.go
@@ -105,6 +105,7 @@ var DeployTestingInfrastructureFlag = false
 var PathToTestingInfrastrucureManifests = ""
 
 func init() {
+	kubecli.Init()
 	flag.StringVar(&KubeVirtUtilityVersionTag, "utility-container-tag", "", "Set the image tag or digest to use")
 	flag.StringVar(&KubeVirtVersionTag, "container-tag", "latest", "Set the image tag or digest to use")
 	flag.StringVar(&KubeVirtVersionTagAlt, "container-tag-alt", "", "An alternate tag that can be used to test operator deployments")

